### DISCUSSION TITLE
GLIBC_2.28

### DISCUSSION THREAD

**Discussion by JavierPalina** [vercel/community#2258] [Help] [answered]
### Summary

Hi! i need help with an issue in Vercel, i have a project in NextJS with canvas, and i receive this error

Error: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by /vercel/path0/node_modules/canvas/build/Release/libglib-2.0.so.0)

someone can help me please?
![image](https://user-images.githubusercontent.com/83319652/235546290-62c8000b-4edd-4cd2-bf0d-f4c56a7d2d97.png)

**Comment by i-am-the-slime** [comment]
I have the same issue with a different library. I think the underlying reason is that Vercel uses the old Amazon Linux 2. If there was a way to use 2022 or 2023 there'd be a more recent version of glibc this should work.

**Comment by Pieparker** [comment, marked as answer]
Hey there! A few workarounds. Either:
- compile the modules from source
- consider an earlier version of Node (IIRC this is a Node 18 quirk)
- have `"canvas": "^2.11.0"` in package.json, and [update your build command as necessary](https://github.com/vercel/vercel/issues/3460#issuecomment-583072848)
